### PR TITLE
Afsaneh/onboarding

### PR DESCRIPTION
**Pull request by afsaneh-rezaeinejad-deriv**

### CODE REVIEW

File: cypress/e2e/smoke/Wallets/addTradingAccount.cy.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import '@testing-library/cypress/add-commands'
+
+describe('QATEST-156146 - Add Trading account', () => {
+  beforeEach(() => {
```
Comment (abuzar-deriv) [inline cypress/e2e/smoke/Wallets/addTradingAccount.cy.js:4]: This test is archived. Sould be `QATEST-142456`. Can you please check once?

File: cypress/e2e/smoke/Wallets/onboarding.cy.js [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+    title: "Trader's Hub tour",
+    description: 'Press here to repeat this tour.',
+  },
+]
+
+const setupTest = () => {
+  cy.findByTestId('dt_traders_hub_onboarding_icon').click()
+  cy.contains('USD Wallet', { timeout: 10000 }).should('exist')
+  cy.get('#react-joyride-portal').should('exist')
+  cy.get('[data-test-id="spotlight"]').should('exist')
```
Comment (abuzar-deriv) [inline cypress/e2e/smoke/Wallets/onboarding.cy.js:73]: Why not `findByTestId()`?

File: cypress/e2e/smoke/Wallets/onboarding.cy.js [hunk 3]
```diff
@@ -0,0 +1,300 @@
...
+    .find('button')
+    .then((buttons) => {
+      const buttonCount = buttons.filter(
+        (index, button) => Cypress.$(button).text().trim() === 'Add'
+      ).length
+      return buttonCount === 0
+    })
+}
+
+describe('QATEST-98504 - User Onboarding on Desktop for Fiat Wallets and QATEST-98504  Launch onboarding from different pages', () => {
```
Comment (abuzar-deriv) [inline cypress/e2e/smoke/Wallets/onboarding.cy.js:105]: Same test ID. Why mention it twice?

File: cypress/e2e/smoke/Wallets/onboarding.cy.js [hunk 4]
```diff
@@ -0,0 +1,300 @@
...
+  beforeEach(() => {
+    cy.c_login({ user: 'walletloginEmail' })
+  })
+
+  it('User onboarding for desktop', () => {
+    cy.c_visitResponsive('/', 'large')
+    const walletAdded = allWalletAdded()
+    cy.contains('Wallet', { timeout: 10000 }).should('exist')
```
Comment (abuzar-deriv) [inline cypress/e2e/smoke/Wallets/onboarding.cy.js:113]: Question: `findByText()` didn't work?

File: cypress/e2e/smoke/Wallets/onboarding.cy.js [hunk 5]
```diff
@@ -0,0 +1,300 @@
...
+    desktopSteps.forEach((step, index) => {
+      if (index === 3 && walletAdded) return
+      checkStep(
+        step.backButtonExists,
+        step.nextButtonName,
+        step.title,
+        step.description
+      )
+    })
+    cy.get('[data-test-id="spotlight"]').should('not.exist')
```
Comment (abuzar-deriv) [inline cypress/e2e/smoke/Wallets/onboarding.cy.js:124]: Please try using `findByTestId()`.

File: cypress/e2e/smoke/Wallets/onboarding.cy.js [hunk 6]
```diff
@@ -0,0 +1,300 @@
...
```
Comment (abuzar-deriv) [inline cypress/e2e/smoke/Wallets/onboarding.cy.js:142]: Same here. Please try using `findByTestId()`.

File: cypress/e2e/smoke/Wallets/onboarding.cy.js [hunk 7]
```diff
@@ -0,0 +1,300 @@
...
```
Comment (abuzar-deriv) [inline cypress/e2e/smoke/Wallets/onboarding.cy.js:159]: Please try using `findByTestId()`.

File: cypress/e2e/smoke/Wallets/onboarding.cy.js [hunk 8]
```diff
@@ -0,0 +1,300 @@
...
```
Comment (abuzar-deriv) [inline cypress/e2e/smoke/Wallets/onboarding.cy.js:178]: Please try using `findByTestId()`.

File: cypress/e2e/smoke/Wallets/onboarding.cy.js [hunk 9]
```diff
@@ -0,0 +1,300 @@
...
+  })
+  it('User onboarding from BTC wallet cashier', () => {
+    cy.c_visitResponsive('/', 'large')
+    cy.contains('Wallet', { timeout: 10000 }).should('exist')
+    cy.c_switchWalletsAccount('BTC')
+    const walletAdded = allWalletAdded()
+    cy.findByText('Deposit').click()
+    // cy.get('canvas').should('be.visible')
```
Comment (abuzar-deriv) [inline cypress/e2e/smoke/Wallets/onboarding.cy.js:167]: Please remove this comment.

File: cypress/e2e/smoke/Wallets/onboarding.cy.js [hunk 10]
```diff
@@ -0,0 +1,300 @@
...
+    desktopSteps.forEach((step, index) => {
+      if (index === 3 && walletAdded) return
+      checkStep(
+        step.backButtonExists,
+        step.nextButtonName,
+        step.title,
+        step.description
+      )
+    })
+    cy.get('[data-test-id="spotlight"]').should('not.exist')
```
Comment (abuzar-deriv) [inline cypress/e2e/smoke/Wallets/onboarding.cy.js:195]: Please try using `findByTestId()`.

File: cypress/e2e/smoke/Wallets/onboarding.cy.js [hunk 11]
```diff
@@ -0,0 +1,300 @@
...
```
Comment (abuzar-deriv) [inline cypress/e2e/smoke/Wallets/onboarding.cy.js:214]: Please try using `findByTestId()`.

File: cypress/e2e/smoke/Wallets/onboarding.cy.js [hunk 12]
```diff
@@ -0,0 +1,300 @@
...
+    mobileSteps.forEach((step, index) => {
+      if (index === 3 && walletAdded) return
+      checkStep(
+        step.backButtonExists,
+        step.nextButtonName,
+        step.title,
+        step.description
+      )
+    })
+    cy.get('[data-test-id="spotlight"]').should('not.exist')
```
Comment (abuzar-deriv) [inline cypress/e2e/smoke/Wallets/onboarding.cy.js:229]: Please try using `findByTestId()`.

File: cypress/e2e/smoke/Wallets/onboarding.cy.js [hunk 13]
```diff
@@ -0,0 +1,300 @@
...
```
Comment (abuzar-deriv) [inline cypress/e2e/smoke/Wallets/onboarding.cy.js:247]: Please try using `findByTestId()`.

**Review by abuzar-deriv — CHANGES_REQUESTED** [review]
There're multiple uses of `cy.get('data-test-id'...` and `cy.contains()` which should be replaced with `findByTestId()` and `findByText()` respectively. Also left some other comments.

**Review by bilalmajid-deriv — APPROVED** [review]
LGTM apart from  changes ahighlighted by @abuzar-deriv

File: cypress/e2e/smoke/Wallets/onboarding.cy.js (hunk 2 above)
Comment (abuzar-deriv) [reply cypress/e2e/smoke/Wallets/onboarding.cy.js:73]: Discussed with @afsaneh-rezaeinejad-deriv. Issue in how this locator is added, might need dev to update it later.

**Review by abuzar-deriv — APPROVED** [review]
LGTM

**Review by markodonnell-deriv — CHANGES_REQUESTED** [review]
I can't see this QATEST-142456 in the Test Case Bank: https://app.clickup.com/20696747/v/l/kqknb-583515

**Review by markodonnell-deriv — APPROVED** [review]
Looks good

**Pull request merged**